indexer/index_user.go

```diff
@@ -1,67 +1,41 @@
 package indexer
 
 import (
-	"encoding/json"
-	"strings"
+	"context"
+	"crypto/ecdsa"
+	"encoding/base64"
 
-	core_proto "github.com/AudiusProject/audiusd/pkg/api/core/v1"
 	"github.com/jackc/pgx/v5"
-)
-
-func (ci *CoreIndexer) createUser(txInfo TxInfo, em *core_proto.ManageEntityLegacy) error {
-	var metadata GenericMetadata
-	json.Unmarshal([]byte(em.Metadata), &metadata)
+	"go.uber.org/zap"
 
-	args := pgx.NamedArgs{
-		"user_id":              em.EntityId,
-		"handle_lc":            strings.ToLower(metadata.Data["handle"].(string)),
-		"is_current":           true,
-		"is_verified":          false,
-		"created_at":           txInfo.timestamp,
-		"updated_at":           txInfo.timestamp,
-		"has_collectibles":     false,
-		"txhash":               txInfo.txhash,
-		"is_deactivated":       false,
-		"is_available":         true,
-		"is_storage_v2":        false,
-		"allow_ai_attribution": false,
-	}
-
-	for k, v := range metadata.Data {
-		if k == "events" {
-			continue
-		}
-		args[k] = v
-	}
-
-	return ci.doInsert("users", args)
-}
+	corev1 "github.com/AudiusProject/audiusd/pkg/api/core/v1"
+	core_config "github.com/AudiusProject/audiusd/pkg/core/config"
+	"github.com/AudiusProject/audiusd/pkg/core/server"
+	"github.com/ethereum/go-ethereum/crypto"
+)
 
-func (ci *CoreIndexer) updateUser(txInfo TxInfo, em *core_proto.ManageEntityLegacy) error {
-	var metadata GenericMetadata
-	json.Unmarshal([]byte(em.Metadata), &metadata)
+func (ci *CoreIndexer) SetPubkeyForUser(dbTx pgx.Tx, logger *zap.Logger, userId int32, pubkey *ecdsa.PublicKey) {
```

Comment:
super nitpicky but I've been starting to think we might want to adopt a "private by default" standard with our functions rather than the public one we've been using

Reply:
Oh yeah I'm fine with making it private by default. Was tab-completing a lot of it :-)